rcl/src/rcl/wait.c

```diff
@@ -267,53 +210,47 @@ rcl_wait_set_get_allocator(const rcl_wait_set_t * wait_set, rcl_allocator_t * al
-    sizeof(rcl_ ## Type ## _t *) * wait_set->size_of_ ## Type ## s); \
-  wait_set->impl->Type ## _index = 0;
+  do { \
+    memset( \
+      (void *)wait_set->Type ## s, \
+      0, \
+      sizeof(rcl_ ## Type ## _t *) * wait_set->size_of_ ## Type ## s); \
+    wait_set->impl->Type ## _index = 0; \
+  } while (false)
 
 #define SET_CLEAR_RMW(Type, RMWStorage, RMWCount) \
-  /* Also clear the rmw storage. */ \
-  memset( \
-    wait_set->impl->RMWStorage, \
-    0, \
-    sizeof(void *) * wait_set->impl->RMWCount); \
-  wait_set->impl->RMWCount = 0;
+  do { \
+    /* Also clear the rmw storage. */ \
+    memset( \
+      wait_set->impl->RMWStorage, \
+      0, \
+      sizeof(void *) * wait_set->impl->RMWCount); \
+    wait_set->impl->RMWCount = 0; \
+  } while (false)
 
 #define SET_RESIZE(Type, ExtraDealloc, ExtraRealloc) \
-  RCL_CHECK_ARGUMENT_FOR_NULL(wait_set, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator()); \
-  RCL_CHECK_FOR_NULL_WITH_MSG( \
-    wait_set->impl, "wait set is invalid", \
-    return RCL_RET_WAIT_SET_INVALID, rcl_get_default_allocator()); \
-  if (size == wait_set->size_of_ ## Type ## s) { \
-    return RCL_RET_OK; \
-  } \
-  rcl_allocator_t allocator = wait_set->impl->allocator; \
-  wait_set->size_of_ ## Type ## s = 0; \
-  wait_set->impl->Type ## _index = 0; \
-  if (size == 0) { \
-    if (wait_set->Type ## s) { \
-      allocator.deallocate((void *)wait_set->Type ## s, allocator.state); \
-      wait_set->Type ## s = NULL; \
+  do { \
+    rcl_allocator_t allocator = wait_set->impl->allocator; \
+    wait_set->size_of_ ## Type ## s = 0; \
+    wait_set->impl->Type ## _index = 0; \
+    if (0u == Type ## s_size) { \
+      if (wait_set->Type ## s) { \
+        allocator.deallocate((void *)wait_set->Type ## s, allocator.state); \
+        wait_set->Type ## s = NULL; \
+      } \
+      ExtraDealloc \
+    } else { \
+      wait_set->Type ## s = (const rcl_ ## Type ## _t **)allocator.reallocate( \
+        (void *)wait_set->Type ## s, sizeof(rcl_ ## Type ## _t *) * Type ## s_size, \
+        allocator.state); \
+      RCL_CHECK_FOR_NULL_WITH_MSG( \
+        wait_set->Type ## s, "allocating memory failed", \
+        return RCL_RET_BAD_ALLOC, wait_set->impl->allocator); \
+      memset((void *)wait_set->Type ## s, 0, sizeof(rcl_ ## Type ## _t *) * Type ## s_size); \
+      wait_set->size_of_ ## Type ## s = Type ## s_size; \
+      ExtraRealloc \
     } \
-    ExtraDealloc \
-  } else { \
-    wait_set->Type ## s = (const rcl_ ## Type ## _t **)allocator.reallocate( \
-      (void *)wait_set->Type ## s, sizeof(rcl_ ## Type ## _t *) * size, allocator.state); \
-    RCL_CHECK_FOR_NULL_WITH_MSG( \
-      wait_set->Type ## s, "allocating memory failed", \
-      return RCL_RET_BAD_ALLOC, wait_set->impl->allocator); \
-    memset((void *)wait_set->Type ## s, 0, sizeof(rcl_ ## Type ## _t *) * size); \
-    wait_set->size_of_ ## Type ## s = size; \
-    ExtraRealloc \
-  } \
-  return RCL_RET_OK;
+  } while (false)
```

Comment:
Wouldn't wrapping the block in curly braces be sufficient if you need to contain the scope of the variables?

Reply:
Same answer as https://github.com/ros2/rcl/pull/285#discussion_r212452894